Demon Antag [WIP]

lordpidey opened the pull request:
Hello, this is a major WIP, with only some basic outlines done.  But the idea is that it's a new antag type, probably either a traitor/demon gamemode (similar to traitor/changling), or as a random event (similar to revenant)
:cl:
rscadd: Infernal demons have been seen onboard our spacestations, offering great boons in exchange for souls.
rscadd: Employees are reminded that their souls already belong to nanotransen.  If you have sold your soul in error, a lawyer or head of personnel can help return your soul to Nanotransen by hitting you with your employment contract.
rscadd: Nanotransen headquarters will be bluespacing employment contracts into the Lawyer's office filing cabinet when a new arrival reaches the station.  It is recommended that the lawyer create copies of some of these for safe keeping.
rscadd: Due to the recent demonic incursions, the station library has been equipped with a demonomicon to help research demonic weaknesses.  Please note that reading this book may have unintended side effects.  Also note, you must spell the demon's name exactly, as there are countless demons with similar names.
rscadd: When a demon dies, if the proper banishment ritual is not performed on it's remains, the demon will revive itself at the cost of some of it's power.  The banishment ritual is described in the demonomicon.
rscadd: When a demon gains enough souls, It's form will mutate to a more demonic looking form.  The Arch-demon form is known to be on par with an ascended shadowling in power.
/:cl:

[x] create contracts
├ [X] create filing cabinet with employment contracts
├ [x] allow copier to copy employment contracts
├ [X] Add boons to willingly signed contracts
└ [x] Lawyer/HoP bashing someone with employment contract can return soul.

 [X] Add demon antag.
├ [X] Add demon objectives.
├ [X]  Add demon name generator.
├ [X] Add demon bans.  (A list of things the demon cannot/must do, similar to silicon laws)
│ ├ [X] Make demon's bane functional
│ ├ [X] Make ability to research demon's ban.
│ └ [X] Give librarian ability to research the demon's weaknesses.
│    └ [X]  Researching demons can have detrimental effects.
├ [X] Add demon spells/powers
├ [X] Sprites?
├ ~~[ ] Ability to force people to sign a contract, for less of a benefit than a willing signature.~~
│   That seems like a dumb idea.  Scratching it for now.
├ [X] Add admintools for adding/removing demons.
├ [\] Demons increase in power as they gain more souls
├ [\] Final form, Archdemon, is an instant shuttle call (akin to ascended shadowling)
│  └ [ ] When a demon reaches archdemon form, all souls that he owns are transformed into slaughter
│          demons.
└ [ ] Demons lose power if they die, but resurrect.
   └ [ ] Add rituals to prevent demonic resurrection.

 [X] Add Sintouched antags.
└ [X] Intentional exposure to demonic information causes sintouched.

~~[ ] Add Hell~~
~~├ [ ] Make the damned enter hell upon death.~~
~~├ [ ] add "fun" things to hell.~~
~~└ [ ] Add ways to break out of hell, with outside help.~~
~~   └ [ ] Science can make portal to hell.~~
~~      └ [ ] Portal can backfire badly.~~
Yeah, I'm going to skip making hell for now.

Misc stuff:
[ ] Add toy demonologist to arcade rewards.

Suggestions I've heard so far that I plan on implementing.
 [X] Move demonomicon from ground.
 [ ] Alter the "magic" contract.
 [X] Make the employment cabinet look more important.

lordpidey committed:
Beginning to work on demon gamemode.  Specifically, starting on contracts.

lordpidey committed:
Merge remote-tracking branch 'tgrepo/master' into Demon

lordpidey committed:
Merge remote-tracking branch 'TgRepo/master' into demon

lordpidey committed:
Merge commit '64574799345e8da49a7ac75fd3675089385e1f97' into demon

lordpidey committed:
Made outline for returning a soul with employment contract.

lordpidey committed:
Photocopier can now copy employment contracts, and blows up upon inserting an infernal contract.

lordpidey committed:
Merge remote-tracking branch 'tgrepo/master' into demon

lordpidey committed:
Fixed 2x compile errors.

lordpidey committed:
Fixed compile errors, it now compiles.

TechnoAlchemisto commented:
Seems interesting, don't fuck it up!

AlexCrimson commented:
I doubt you would need to force people to sign contracts. Most would willingly do it. Depends what they get in return. Should definitely give the Demon some ability to contact people before their death and heal/save them somehow in return for their soul.

Maybe some ritual to summon the thing? May need to be randomized, as people would metagame it.

TrustyGun commented:
> [ ] Portal can backfire badly.

finally we can have doom in spess

Please make this work, this looks like it could be really fun

MMMiracles commented:
poor incoming

RemieRichards commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,140 @@
+		M.removeDemonBoons()
+		return
+	else
+		if(ishuman(M) && !istype(M:head, /obj/item/clothing/head/helmet))
```
The colon operator ```:``` is not allowed, please typecast and use the dot operator ```.``` instead.

RemieRichards commented:
I'm VERY interested in this.

Jazaen commented:
[ ] Science can make portal to hell.

Please, could you add this to some other job? Maybe chaplain or some other service related one? So much stuff is already gated behind science that adding more to them seems like a bad idea.

neersighted commented:
This looks interesting. Inspired by that slaughter daemon forum thread?

lordpidey commented:
@neersighted, no, it's not, this is actually an older idea that I had.  I don't frequent the forums.

duncathan commented:
this will DEFINITELY conflict with datum antags. 
@Incoming5643 should this be DNM until after you're ready, or would you rather deal with the conflicts yourself? Your decision.

lordpidey commented:
@duncathan, it will take me a while to get the rest of the stuff done,  Perhaps it would be best for ME to wait for datum antags to finish?

JJRcop commented:
@lordpidey I'm not @duncathan, but I think that it wouldn't be a bad idea. You can finish it, then wait for datum antags to come out, then work on converting it.

Incoming5643 commented:
If it's a self contained simple animal antag that's easy for me to convert. Otherwise yeah might wanna wait.

PKPenguin321 commented:
This actually sounds really, really cool. :+1:

Supermichael777 commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,140 @@
+/obj/item/weapon/paper/contract/infernal/suicide_act(mob/user)
+	if(signed && (user == target))
+		user.say("OH GREAT INFERNO!  I DEMAND YOU COLLECT YOUR BOUNTY IMMEDIATELY!")
+		user.visible_message("<span class='suicide'>[user] holds up a contract claiming his soul, then immediately catches fire.  Their corpse smelling of brimstone.</span>")
```
smells, also this is a visible mesage that last sentence doesn't add anything and doesnt add anything

Supermichael777 commented:
picks between the clown and roundstart lawyer because you kow the clown is the devil

Zilenan91 commented:
So is this anything like the Fallen from Lifeweb? If it's literally that then this would be awesome.

What the Fallen is is an invisible fleshy monstrosity that can only be seen by one person. The person it can be seen by is given an offer, a great boon to them, at the cost of their soul. They eventually grow in power via these deals and kill the angel that is also there that is attempting to stop them, thus taking over the world.

Alek2ander commented on code/modules/paperwork/contract.dm:
```diff
@@ -0,0 +1,140 @@
+	if(istype(P, /obj/item/weapon/pen) || istype(P, /obj/item/toy/crayon))
+		if(user.IsAdvancedToolUser())
+			if(user.hasSoul)
+				user << "<span class='notice'>You quickly scrawl your name on the contract</span>"
```
Aren't you supposed to sign these with blood?

Alek2ander commented:
>>[ ] Science can make portal to hell.

>Please, could you add this to some other job? Maybe chaplain or some other service related one? So much stuff is already gated behind science that adding more to them seems like a bad idea.

I think that the portal should _be_ a backfire if science has gone too far (experimentor spam, telescience abuse, etc) with lethal consequences.

lordpidey commented:
@Zilenan91, I've never played lifeweb, so I can't say for sure, but yes I was planning on the demon getting stronger the more souls he gets.

lordpidey committed:
Added demon name/weakness generator.  Also added a toy that uses the name/weakness generator.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

RemieRichards commented on code/game/gamemodes/demon/demoninfo.dm:
```diff
@@ -0,0 +1,99 @@
+#define banesilver 0
```
typically defines are capitalised.

#define BANESILVER 0

lordpidey committed:
Capitalized the defines in demoninfo.dm

Zilenan91 commented:
Will the demon be a normal crew member just running around or will he be like a revenant who's only visible to one person?

lordpidey commented:
I was thinking normal crew member at round start, but as he gets more souls, becomes gradually more and more demonic looking.

Zilenan91 commented:
If he's a normal crew member I can 100% guarantee he'll get validded instantly. He should probably be like a revenant and becomes visible to people when he's making a deal.

lordpidey commented:
I mean, I'm planning on giving him demonic powers to help him escape.  He'll be stronger than a normal crew member, by a good deal.  Without knowing his weakness, he will be quite difficult to kill.

AlexCrimson commented:
People will meta the shit out of his weaknesses unless you randomise them each round or something silly like that.

lordpidey commented:
I've already made a random weakness generator (though the weaknesses don't do anything at the moment).  The idea is if the crew can somehow figure out the true name of the demon (likely by finding one of his contracts), they can research his weakness.

KorPhaeron commented:
>Hell

What turfs etc is hell going to consist of?

lordpidey commented:
Honestly, I'm not sure about Hell's tileset yet, but I am open to suggestions.  My CURRENT idea for breaking out of hell is as follows:  (Numbers are arbitrary that I pulled out of my ass).

If you are hellbound, and die, you will instantly respawn in hell, naked, with negative twenty "redemption" points.  Every so often, you will be thrown into a trial.  If you pass the trial, you get ten redemption points.  If you die, you lose fifteen (minimum negative thirty).
Every time you pass a trial you have redemptionPoint% chance of going to purgatory.

Escaping from purgatory is easy with outside help, just have someone zap you up with the gateway. (or something similarly easy).

Incursions into hell itself will also be possible, but far more difficult, and possible to backfire.

lordpidey committed:
Added Demonomicon book, which can lookup demon's weaknesses.

Added outline for bad things happening to those with low willpower (mostly clowns) who read the Demonomicon.

lordpidey committed:
Demonomicon progress bar now works properly.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Added pitchfork weapon,
added codersprite for demonomicon (Might actually keep this sprite)
added codersprite for pitchfork (Needs work)

lordpidey committed:
Added basic demon spells/powers

lordpidey committed:
Added boons for some contract types.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	code/modules/mob/living/carbon/human/human.dm
#	code/modules/mob/living/living.dm

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Becoming sin-touched now adds an appropriate abductee-esque objective.
Made prestige contract more robust.

lordpidey committed:
Fixed indentation.

lordpidey committed:
Added employment contract cabinet.  It partially works, for some reason it's not making contracts for late arrivals correctly, BUT it makes contracts correctly for people that arrived before it was made.
Fixed envy sintouch requirement
Added admintools for looking at sintouched.
Added end round report for sintouched.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Fixed employment contracts not spawning with a name.
Removed debug message.

Cheridan commented:
>demon antag

but we already have a demon antag.

PKPenguin321 commented:
This is a friendlier, not-slaughter demon. He went to demonic law school, whereas the slaughter demons were the jocks.

Shadowlight213 commented:
@Pkpenguin321 
>lawyers being friendly
>less vicious than jocks

Umm. You sure you don't have it backwards?

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	icons/mob/inhands/items_lefthand.dmi
#	icons/obj/weapons.dmi

Solved by just throwing out my shitty codersprites.

lordpidey committed:
People who've sold their souls cannot be resurrected now, even by magic.

lordpidey committed:
Adds preliminary demon objectives.  STill need to add checks for greentext.

lordpidey committed:
Fixes compile error for ministation.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Added admintools for adding a demon (they don't work yet.)
Added round-end reports for demons.
Now keeps track of which demon owns which soul.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	code/modules/spells/spell_types/ethereal_jaunt.dm

lordpidey committed:
Removed slavery contract
Contracts now use demon's TRUENAME
Added boon for contract of knowledge
Added completion check for ensure number of sintouched.

lordpidey committed:
Moved soul ownership to the mind datum, instead of the mob itself.

lordpidey committed:
Slightly reduces memory consumption by not remembering irrelevant details for non-existant demons upon researching.

lordpidey committed:
Implemented demon bane functionality.

lordpidey committed:
Admins can now set people to be demons manually.

lordpidey committed:
Fixed salt and flashes causing spells to be locked indefinately.

lordpidey committed:
Pitchforks no longer burn demons on holding.
Demonomicon can't be read multiple times simultaniously.
Fixed summon contract spell icon.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Buffed flashing demon weakness.
Nerfed salt demon weakness.

lordpidey committed:
Added terrible, terrible codersprites for pitchforks.

lordpidey committed:
Placed demonomicon on some maps.

lordpidey committed:
Can now select which type of contract is summoned.

lordpidey committed:
Fixes salt/flash banes.
Gives summon pitchfork a proper icon.
Ethereal jaunt no longer blinds.
Fixes pitchfork sprites.... They're still terrible though.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	code/modules/spells/spell_types/ethereal_jaunt.dm

lordpidey committed:
Fixed showing demon's objectives.

lordpidey committed:
Fixes revival contracts
Demon information is now stored in notes.

lordpidey committed:
Fixes compile error.

lordpidey committed:
Now stores the CORRECT description of demoninfo in notes.
Removed shadowling hud icon from demons.

lordpidey committed:
Adds demonomicon and employment contract cabinets to maps.

lordpidey commented:
Aright, this should be ready for a preliminary test round or two.

ChangelingRain commented on code/game/gamemodes/miniantags/revenant/revenant.dm:
```diff
@@ -318,13 +318,13 @@
 
 
 /mob/living/simple_animal/revenant/attackby(obj/item/W, mob/living/user, params)
-	if(istype(W, /obj/item/weapon/nullrod))
```
This'll trigger on everything? I don't think you meant to do that.

lordpidey replied:
Yeah, no idea how that happened.  Good catch.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Reverting a change I accidentally made to revenants.
No idea how this got in there, probably I screwed up on  merge

lordpidey committed:
Fixes compile error from improper capitalization.

lordpidey commented:
Okay, I had the first test round of this happen, and I got some good feedback about it.  I'll weaken/remove the contract for magic, and I'll make additional demonomicons orderable from cargo.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

coalescentwinter commented:
To prevent confusion with the actual slaughter demon antag, I propose renaming to devil.

RemieRichards commented:
@coalescentwinter there's more than one concept of Demon, it's fine.

KorPhaeron commented:
Could the chaplains demonomicon or whatever not start on the floor? Looks a bit silly

KorPhaeron commented on code/_onclick/item_attack.dm:
```diff
@@ -25,7 +25,13 @@
+			var/mob/living/carbon/human/H = user
+			if(istype(H.w_uniform, /obj/item/clothing/under/color/white))
+				damage *= 3
+	apply_damage(damage, I.damtype, def_zone)
```
>// Sorry for the snowflake check.

Well you already know what is wrong with all this then. Checks every single time any mob is hit by anything.

RemieRichards replied:
Yeah this can't happen, not like this.

PKPenguin321 replied:
if i'm reading this right, what you could do is make white clothing have a /hit_reaction() like the armor in #13522
unless what you're trying to do is make it so people with white clothes do more damage to demons. then i have no idea

lordpidey replied:
Yeah, it's making white clothed people do more damage to demons with the appropriate bane.  I'm still wracking my brain for a better way to do this.  I'm thinking add a whiteness value to clothing, and then add a check_weakness method to mob/living?

RemieRichards replied:
@lordpidey I would give mobs some kind of "attack intercept" datum, which if returns 1 then normal attackby() isn't called. This attack intercept would allow demon humans to react differently to attacks without it being a snowflake addition, it'd be a generic "attack intercept" system.

KorPhaeron commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -605,8 +605,17 @@
+		if(holder && holder.chem_temp < 100) // COLD iron.
+			M.reagents.add_reagent("toxin", reac_volume)
+	..()
+
```
All these chem checks are pretty snowflake as well. Maybe make the demon its own mutant race (subtype of human) and check the chems there?

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	_maps/map_files/AsteroidStation/AsteroidStation.dmm
#	_maps/map_files/DreamStation/dreamstation04.dmm
#	_maps/map_files/EfficiencyStation/EfficiencyStation.dmm
#	_maps/map_files/TgStation/tgstation.2.1.3.dmm
#	code/modules/mob/living/carbon/human/human.dm
#	icons/mob/inhands/items_lefthand.dmi
#	icons/mob/inhands/items_righthand.dmi

Human.dm was manually merged, the rest simply took the trunk's

lordpidey committed:
Re-factors code to be less snowflake.  Still a lot of work to do on this front, but it's a start.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	icons/mob/inhands/items_lefthand.dmi
#	icons/mob/inhands/items_righthand.dmi
#	icons/obj/weapons.dmi

lordpidey committed:
Illogical contracts no longer work.  (Such as a drone trying to get hulkism)

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	icons/mob/actions.dmi

lordpidey committed:
Fixes harvest bane, adds outline for demonic upgrades with souls.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
I don't remember what I changed this commit, I'm just commiting so I can work on this on another computer easier.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

lordpidey committed:
Added sprites for employment cabinet, and arch demon.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	code/modules/mob/living/carbon/human/human.dm

lordpidey committed:
Contracts slightly re-factored to work with minds instead of mobs.

lordpidey committed:
refactors contracts to use minds instead of mobs.  Makes more sense this way.

lordpidey committed:
Merge remote-tracking branch 'Tgstation/master' into demon

# Conflicts:
#	icons/obj/bureaucracy.dmi

lordpidey committed:
Fixes compile error.

lordpidey commented:
Closed to re-make PR, see #15794 for new PR
